elide-async/src/main/java/com/yahoo/elide/async/hooks/TableExportHook.java

```diff
@@ -15,29 +15,69 @@
 import com.yahoo.elide.core.security.ChangeSpec;
 import com.yahoo.elide.core.security.RequestScope;
 
+import org.apache.http.client.utils.URIBuilder;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URISyntaxException;
 import java.util.Optional;
 
 /**
  * LifeCycle Hook for execution of TableExpoer.
  */
+@Slf4j
 public class TableExportHook extends AsyncAPIHook<TableExport> {
+    private boolean baseURLPresent = true;
 
-    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds) {
+    public TableExportHook (AsyncExecutorService asyncExecutorService, Integer maxAsyncAfterSeconds,
```

Comment:
baseUrlPresent is an odd contract.  Also, it doesn't really belong in the hook.

baseUrl can be extracted from the requestScope.elideSettings.

Constructing the downloadURL should be a utility function that is shared with the export controller and this class.